src/class_sqlThread.py

```diff
@@ -17,128 +17,100 @@
         except Exception as err:
-            if str(err) == 'table inbox already exists':
-                logger.debug('Database file already exists.')
-
-            else:
-                sys.stderr.write(
-                    'ERROR trying to create database file (message.dat). Error message: %s\n' % str(err))
-                os._exit(0)
-
-        # If the settings version is equal to 2 or 3 then the
-        # sqlThread will modify the pubkeys table and change
-        # the settings version to 4.
-        settingsversion = config.getint(
-            'bitmessagesettings', 'settingsversion')
-
-        # People running earlier versions of PyBitmessage do not have the
-        # usedpersonally field in their pubkeys table. Let's add it.
-        if settingsversion == 2:
-            item = '''ALTER TABLE pubkeys ADD usedpersonally text DEFAULT 'no' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-            self.conn.commit()
-
-            settingsversion = 3
-
-        # People running earlier versions of PyBitmessage do not have the
-        # encodingtype field in their inbox and sent tables or the read field
-        # in the inbox table. Let's add them.
-        if settingsversion == 3:
-            item = '''ALTER TABLE inbox ADD encodingtype int DEFAULT '2' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-
-            item = '''ALTER TABLE inbox ADD read bool DEFAULT '1' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-
-            item = '''ALTER TABLE sent ADD encodingtype int DEFAULT '2' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-            self.conn.commit()
-
-            settingsversion = 4
-
-        config.set(
-            'bitmessagesettings', 'settingsversion', str(settingsversion))
-        config.save()
-
-        helper_startup.updateConfig()
-
-        # From now on, let us keep a 'version' embedded in the messages.dat
-        # file so that when we make changes to the database, the database
-        # version we are on can stay embedded in the messages.dat file. Let us
-        # check to see if the settings table exists yet.
-        item = '''SELECT name FROM sqlite_master WHERE type='table' AND name='settings';'''
-        parameters = ''
-        self.cur.execute(item, parameters)
+            logger.debug(
+                'ERROR trying to initialize database. Error message: %s\n', str(err))
+
+    @property
+    def sql_schema_version(self):
+        """
+            Getter for get current schema version
+        """
+        return self.__get_current_settings_version()
+
+    @sql_schema_version.setter
+    def sql_schema_version(self, setter):
+        """
+            Update version with one level
+        """
+        if setter:
+            query = "UPDATE settings SET value=CAST(value AS INT) + 1 WHERE key = 'version'"
```

Comment:
we should have a test of what happens if there is legacy string in the database when retrieving the value